openfe_gromacs/protocols/gromacs_md/md_methods.py

```diff
@@ -0,0 +1,649 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+"""Gromacs MD Protocol --- :mod:`openfe_gromacs.protocols.gromacs_md.md_methods`
+================================================================================
+
+This module implements the necessary methodology tools to run an MD
+simulation using OpenMM tools and Gromacs.
+
+"""
+from __future__ import annotations
+
+import logging
+import pathlib
+import uuid
+from collections import defaultdict
+from collections.abc import Iterable
+from typing import Any
+
+import gufe
+import pint
+from gufe import ChemicalSystem, SmallMoleculeComponent, settings
+from openfe.protocols.openmm_utils import (
+    charge_generation,
+    settings_validation,
+    system_creation,
+    system_validation,
+)
+from openfe.protocols.openmm_utils.omm_settings import BasePartialChargeSettings
+from openfe.utils import log_system_probe, without_oechem_backend
+from openff.interchange import Interchange
+from openff.toolkit.topology import Molecule as OFFMolecule
+from openff.units import unit
+from openff.units.openmm import from_openmm, to_openmm
+from openmmtools import forces
+
+from openfe_gromacs.protocols.gromacs_md.md_settings import (
+    EMOutputSettings,
+    EMSimulationSettings,
+    GromacsMDProtocolSettings,
+    IntegratorSettings,
+    NPTOutputSettings,
+    NPTSimulationSettings,
+    NVTOutputSettings,
+    NVTSimulationSettings,
+    OpenFFPartialChargeSettings,
+    OpenMMEngineSettings,
+    OpenMMSolvationSettings,
+)
+
+logger = logging.getLogger(__name__)
+
+
+pre_defined_settings = {
```

Comment:
there's no real rule here, but I like to use all caps for dictionaries defined like this, makes it easier to keep track in your code as "o this is something I set outside of this class".